### PR TITLE
chore(deps): upgrade AWS crates to v0.54.1

### PR DESCRIPTION
**Pull request by neuronull**
This is carry-over from https://github.com/vectordotdev/vector/pull/16364

### CODE REVIEW

File: Cargo.toml
Comment (spencergilbert) [inline Cargo.toml:181]: We'll want to flip these back over to `native-tls`, Jesse had pointed out this mistake after I had switched to the other branch.

File: src/aws/mod.rs
Comment (spencergilbert) [inline src/aws/mod.rs:55]:
```suggestion
        SdkError::ResponseError(err) => {
            check_response(err.raw())
        }
        SdkError::ServiceError(err) => {
            check_response(err.raw())
        }
```

Another clean up from the PR review before

File: src/aws/mod.rs
Comment (spencergilbert) [inline src/aws/mod.rs:58]:
```suggestion
            warn!("AWS returned an unhandled error, retrying request.");
```

Unless you have a better suggestion on this log

File: src/aws/mod.rs
Comment (spencergilbert) [inline src/aws/mod.rs:169]: Everything below here should be reverted, as the existing code was improvements based on PR feedback

File: src/sinks/aws_kinesis/firehose/config.rs
Comment (spencergilbert) [inline src/sinks/aws_kinesis/firehose/config.rs:178]:
```suggestion
        if let SdkError::ServiceError(inner) = error {
            if let PutRecordBatchErrorKind::ServiceUnavailableException(_) = inner.err().kind {
```

File: src/sinks/aws_kinesis/streams/config.rs
Comment (spencergilbert) [inline src/sinks/aws_kinesis/streams/config.rs:185]:
```suggestion
        if let SdkError::ServiceError(inner) = error {
            if let PutRecordsErrorKind::ProvisionedThroughputExceededException(_) = inner.err().kind {
```

File: src/sinks/aws_s3/integration_tests.rs
Comment (spencergilbert) [inline src/sinks/aws_s3/integration_tests.rs:426]:
```suggestion
            SdkError::ServiceError(inner) => match &inner.err().kind {
```

File: src/sinks/s3_common/config.rs
Comment (spencergilbert) [inline src/sinks/s3_common/config.rs:302]:
```suggestion
                SdkError::ServiceError(inner) => match inner.raw().http().status() {
```

File: src/sinks/aws_cloudwatch_logs/request.rs
Comment (spencergilbert) [inline src/sinks/aws_cloudwatch_logs/request.rs:178]:
```suggestion
                                SdkError::ServiceError(inner) => matches!(
                                    inner.err().kind,
```

File: src/sinks/aws_cloudwatch_logs/request.rs
Comment (spencergilbert) [inline src/sinks/aws_cloudwatch_logs/request.rs:153]:
```suggestion
                                SdkError::ServiceError(inner) => matches!(
                                    inner.err().kind,
```

File: src/sinks/aws_cloudwatch_logs/request.rs
Comment (spencergilbert) [inline src/sinks/aws_cloudwatch_logs/request.rs:105]:
```suggestion
                            if let SdkError::ServiceError(inner) = &err {
```

File: src/sinks/aws_cloudwatch_logs/request.rs
Comment (spencergilbert) [inline src/sinks/aws_cloudwatch_logs/request.rs:107]:
```suggestion
                                    inner.err().kind
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Pile of suggestions from changes made during the previous upgrades PR review.

**Comment by spencergilbert** [comment]
Whoopsie 😅 sorry about that.

File: src/aws/mod.rs
Comment (spencergilbert) [inline src/aws/mod.rs:340]: **nitpicking**: What you don't like my use of whitespace?!

File: src/sinks/s3_common/config.rs
Comment (spencergilbert) [inline src/sinks/s3_common/config.rs:302]: **question**: I thought this failed to run due to ownership issues, which is why I ended up using `.raw()`. If that's not the case 🤷 maybe I was thinking of another place...

File: src/sinks/aws_cloudwatch_logs/request.rs
Comment (spencergilbert) [inline src/sinks/aws_cloudwatch_logs/request.rs:233]: **note**: This wasn't strictly part of the upgrade, I just noticed the linked issue had been resolved previously and I wanted to make the world a slightly better place 😄

**Review by spencergilbert — APPROVED** [review]
I'm still a little spooked by the auth changes 😅 but if this passes all the tests...

Worth a second review, imho

File: src/aws/mod.rs
Comment (neuronull) [reply src/aws/mod.rs:340]: wasn't intentional, probably occurred while fixing the merge conflicts.

File: src/sinks/s3_common/config.rs
Comment (neuronull) [reply src/sinks/s3_common/config.rs:302]: Do you mean failed to compile?
Not sure... I didn't face problems with this when I compiled after resolving the merge conflicts.

Comment (spencergilbert) [reply src/sinks/s3_common/config.rs:302]: Word, it might have been a different place that this happened. IIRC there was an issue because `into_raw()` consumes Self.

If it's working here 👍

**Pull request merged**
